src/passes/builtinHandler/blockMethods.ts

```diff
@@ -0,0 +1,54 @@
+import { MemberAccess, Identifier, ExternalReferenceType } from 'solc-typed-ast';
+import { AST } from '../../ast/ast';
+import { ASTMapper } from '../../ast/mapper';
+import { createCairoFunctionStub, createCallToFunction } from '../../utils/functionGeneration';
+import { createUint256TypeName } from '../../utils/nodeTemplates';
+
+/*
+A subpass that replaces the block.timestamp and block.number methods.
+In Solidity these functions return uint256, but in cairo these will return felts.
+Therefore, we wrap the replacement functions in felt_to_uint256 warplib functions.
+*/
+export class BlockMethods extends ASTMapper {
+  visitMemberAccess(node: MemberAccess, ast: AST): void {
```

Comment:
This method needs to recurse if the member access doesn't match the expeceted pattern. Consider `f(block.timestamp).member`